app/assets/javascripts/admin/modules/unpublish-display-conditions.js

```diff
@@ -17,6 +17,25 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
 
   UnpublishDisplayConditions.prototype.init = function () {
     this.initUnpublishTypeListener()
+    this.initUnpublishRedirectListener()
+  }
+
+  UnpublishDisplayConditions.prototype.initUnpublishRedirectListener = function () {
+    var checkbox = this.module.querySelector('.govuk-checkboxes__item input[name="unpublishing[redirect]"]')
+
+    checkbox.addEventListener('change', function (e) {
+      this.showPublicExplanation(!e.target.checked)
+    }.bind(this))
+
+    if (checkbox.checked) {
+      var event = new Event('change')
+      checkbox.dispatchEvent(event)
+    }
+  }
+
+  UnpublishDisplayConditions.prototype.showPublicExplanation = function (checkboxValue) {
+    var display = checkboxValue ? 'block' : 'none'
+    this.module.querySelector('.js-unpublish-withdraw-form__published-in-error div.gem-c-textarea').style.display = display
```

Comment:
We could just have this within the listener as it does one thing, might mean less lines of code.